### PR TITLE
Continuation of #23624

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

See https://github.com/symfony/symfony/pull/23624#discussion_r131539736

cc @chalasr 

Also included service injection for init:acl + set:acl and simplification of lint:xliff + lintt:yaml.

Btw, i think init:acl needs to be renamed to acl:init :)

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php [hunk 1]
```diff
@@ -114,6 +116,9 @@ public function load(array $configs, ContainerBuilder $container)
         // load ACL
         if (isset($config['acl'])) {
             $this->aclLoad($config['acl'], $container);
+        } else {
+            $container->removeDefinition(InitAclCommand::class);
```
Comment (chalasr) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: broken if console not available, should go in the previous if (console.xml loading) and acl loading be moved just before console loading

Comment (ro0NL) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: not sure.. removeDefinition is non-fatal. Did this for translations, routing, etc. as well.

File: src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php [hunk 2]
```diff
@@ -25,11 +27,39 @@
...
      * {@inheritdoc}
+     *
+     * BC to be removed in 4.0
      */
     public function isEnabled()
     {
+        if (null !== $this->connection) {
+            return parent::isEnabled();
+        }
         if (!$this->getContainer()->has('security.acl.dbal.connection')) {
```
Comment (chalasr) [inline src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php]: `if (!$this->connection && ..)` instead?

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php (hunk 1 above)
Comment (chalasr) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: fair

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Bridge/Twig/Command/LintCommand.php [hunk 3]
```diff
@@ -38,16 +38,16 @@ class LintCommand extends Command
...
         if (!$twig instanceof Environment) {
             @trigger_error(sprintf('Passing a command name as the first argument of "%s" is deprecated since version 3.4 and will be removed in 4.0. If the command was registered by convention, make it a service instead.', __METHOD__), E_USER_DEPRECATED);
 
-            $this->setName(null === $twig ? 'lint:twig' : $twig);
+            parent::__construct($twig);
 
             return;
         }
 
+        parent::__construct();
```
Comment (chalasr) [inline src/Symfony/Bridge/Twig/Command/LintCommand.php]: good catch

File: src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php (hunk 2 above)
Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php]: @ro0NL ?

Comment (ro0NL) [inline src/Symfony/Bundle/SecurityBundle/Command/InitAclCommand.php]: Done.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by ro0NL** [comment]
Not sure what appveyor complains about =/

**Comment by nicolas-grekas** [comment]
Thank you @ro0NL.

**Pull request closed without merging**